### ISSUE TITLE
Add ability to claim Survivors for a Team and create a Team

### ISSUE THREAD

**Issue by mawilson** [mawilson/survivor_draft#65]
After implementing https://github.com/mawilson/survivor_draft/pull/64 & Team claiming, I thought it made sense to add the ability to create a team as well. However, after working on this, it seemed a little empty - making a team only for it to be devoid of survivors & requiring admin intervention seemed pointless.

Naturally the next step is to allow teams created to claim Survivors, just as the User could claim a Team. This also allows the draft proper to involve users more naturally. Users can claim the Survivors they draft on the day of the draft themselves thru the website, rather than requiring me to do admin intervention. There are obvious security concerns - one single team could just claim everyone survivor, & then I'd have to clean up afterward. Maybe adding a Boolean to open up claiming survivors would make sense, so that it's only open in a window. But I think the interactivity seems like a good feature.

In short:
* Add ability to claim survivors for a team
* Add a Boolean for this survivor claim window to be open
* Add ability to create a team if a user is not yet associated with a team for a season
* Add ability to unclaim survivors - I think this should also be limited to the claim window Boolean
